### ISSUE TITLE
Web Throttle/SD Card on board

### ISSUE THREAD

**Issue by davidcutting42** [DCC-EX/CommandStation-EX#74]
From #66 - Might host turnout/input/output data, and/or host an instance of exWebThrottle.

**Comment by matt-hu** [comment]
Having an instance of exWebThrottle onboard requires a webserver, something not available without programming the ESP8266, as far as I am aware. Also, the ESP8266 might not be the _fastest_ web server ever... may need an ESP32.

**Comment by Asbelos** [comment]
Web server speed will be pretty irrelevant when you only have to serve a handful of static files once to each of a handful of clients.On 21 Nov 2020 17:50, Matt <notifications@github.com> wrote:
Having an instance of exWebThrottle onboard requires a webserver, something not available without programming the ESP8266, as far as I am aware. Also, the ESP8266 might not be the fastest web server ever... may need an ESP32.

**Comment by matt-hu** [comment]
@Asbelos very true, although I think the major flaw is having to reprogram the ESP every time we release a new version of exWebThrottle. (for more info on exWebThrottle distribution, see DCC-EX/exWebThrottle#15)

**Comment by davidcutting42** [comment]
> the major flaw is having to reprogram the ESP every time we release a new version of exWebThrottle

@matt-hu not if you have an SD card. Then you can just have it serve static files from the SD card. 

I also have a hard time believing that we can't serve exWebThrottle over AT commands. Might be a little hacky, but if you look around with a quick google search there are ways.

**Comment by matt-hu** [comment]
@davidcutting42 Ok, those are good ideas...
I will look into the AT command stuff, but I also think that PWAs should be used as well, so users have a choice.

**Comment by ManiAkasapu** [comment]
Come up best idea Gents..:)

**Comment by davidcutting42** [comment]
@matt-hu I agree that the ideal option would be to port everything over to the ESP32 and serve the webpage up from there. Unfortunately, the ESP would require a lot of changes and bloat in the current firmware. We are considering the option of moving to the ESP32 as our only platform for the v4 release and maintaining backwards compatibility with v3.
